lib/sippy_cup/scenario.rb

```diff
@@ -156,7 +156,9 @@ def invite(opts = {})
       # FIXME: The DTMF mapping (101) is hard-coded. It would be better if we could
       # get this from the DTMF payload generator
       from_addr = "#{@from_user}@#{@adv_ip}:[local_port]"
-      msg = <<-MSG
+      opt_hdrs = (opts.has_key?(:auth_keyword) ? opts[:auth_keyword] : '')
+      + (opts.has_key?(:headers) ? "\n#{opts[:headers]}" : '' ) 
```

Comment:
This line removes behaviour before as it calls `.sub(/\n*\Z/, "\n")`. You should be able to modify the method so you can pass the [authorization] information in the headers option like so:
```
@scenario = SippyCup::Scenario.new 'Invite Test', options do |s|
    s.invite headers: "[authorization username=username password=password]"
end
```
then this makes the auth_keyword option unnecessary